### PR TITLE
Add TS version of coverage command

### PR DESCRIPTION
**Pull request by narehart**
- Removes unmaintained dependency: https://github.com/Connected-Information-systems/diff-test-coverage
- Rewrite coverage in TS
- Add tests for coverage command

### CODE REVIEW

File: pkgs/kona/src/cli/coverage/templates/coverage-files-complete.template [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<% if(Object.keys(coverageByFile).length){ %>Diff coverage statistics:
+------------------------------------------------------------------------------
+|     lines      |    branches    |   functions    |   path
+------------------------------------------------------------------------------
+<% Object.entries(coverageByFile).forEach(coverage => { %>| ${displayCoverage(coverage[1], 'lines', true)} | ${displayCoverage(coverage[1], 'branches', true)} | ${displayCoverage(coverage[1], 'functions', true)} | ${getRelativePath(coverage[1].file)}
+<% }); %>------------------------------------------------------------------------------
+<% } else { %>No coverage information found for any of the new lines in the diff.
+<% } %>
```
Comment (ze-flo) [inline pkgs/kona/src/cli/coverage/templates/coverage-files-complete.template:8]: Line break?

**Review by ze-flo — APPROVED** [review]
Outstanding work! 🔥 🔥 🚀  You should make this your own package. You deserve it! 😉

**Comment by mybrainishuge** [comment]
Looks like this resolves at least 3 vulnerabilities reported in [CORE-1960](https://seismic.atlassian.net/browse/CORE-1960). 🎉

File: pkgs/kona/src/cli/coverage/diff_addition_filter_merge.ts [hunk 2]
```diff
@@ -61,7 +61,7 @@ function mergeChanges(fileDiffs: File[]) {
     const lines = Object.entries(changes)
         .map(entry => {
             const [, change] = entry
-            const line = change.type === 'normal' ? change.ln1 : change.ln
+            const line = change.type === 'normal' ? change.ln2 : change.ln
```
Comment (mybrainishuge) [inline pkgs/kona/src/cli/coverage/diff_addition_filter_merge.ts:64]: What's `ln2` vs `ln1`?

Comment (narehart) [reply pkgs/kona/src/cli/coverage/diff_addition_filter_merge.ts:64]: The `normal` change type is a change that doesn't add or delete any code. It's just something that has moved. `ln1` is the old line it was on and `ln2` is the new line. We want to use the new line when testing.

**Pull request merged**